R/sgdnet.R

```diff
@@ -0,0 +1,103 @@
+#' Fit a Generalized Linear Model with Elastic Net Regularization
+#'
+#' @param x input matrix
+#' @param y response variable
+#' @param family reponse type
+#' @param alpha elastic net mixing parameter
+#' @param lambda regularization strength
+#' @param intercept whether to fit an intercept or not
+#' @param maxit maximum number of effective passes (epochs)
+#' @param standardize whether to standardize `x` or not -- ignored when
+#'   `intercept == FALSE`.
+#' @param thresh tolerance level for termination of the algorithm
```

Comment:
would be nice to have details about how this tolerance is used -- when do we stop? using the difference between gradient norm, objective function values, etc?